deb/openmediavault/workbench/src/app/shared/components/task-dialog-ssh/task-dialog-ssh.component.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * This file is part of OpenMediaVault.
+ *
+ * @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
+ * @author    Volker Theile <volker.theile@openmediavault.org>
+ * @copyright Copyright (c) 2009-2025 Volker Theile
+ *
+ * OpenMediaVault is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * OpenMediaVault is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { marker as gettext } from '@ngneat/transloco-keys-manager/marker';
+import * as _ from 'lodash';
+import { Subscription } from 'rxjs';
+import stripAnsi from 'strip-ansi';
+
+import { format } from '~/app/functions.helper';
+import { Icon } from '~/app/shared/enum/icon.enum';
+import { TaskDialogSshConfig } from '~/app/shared/models/task-dialog-ssh-config.type';
+import { AuthSessionService } from '~/app/shared/services/auth-session.service';
+import { Terminal } from "@xterm/xterm";
+import { FitAddon } from "@xterm/addon-fit";
+import { WebLinksAddon } from '@xterm/addon-web-links';
+import { io, Socket } from 'socket.io-client';
+
+/**
+ * A dialog with 'Start', 'Stop' and 'Cancel' buttons. It will
+ * display the content received by a background task.
+ */
+@Component({
+  selector: 'omv-task-dialog-ssh',
+  templateUrl: './task-dialog-ssh.component.html',
+  styleUrls: ['./task-dialog-ssh.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class TaskDialogSshComponent implements OnInit, OnDestroy {
+  @ViewChild('terminal', { static: true }) terminalDiv!: ElementRef;
+  private socket: Socket;
+  private term: Terminal;
+  private inputBuffer: string = '';
+
+  @Output()
+  readonly finishEvent = new EventEmitter<string>();
+
+  @ViewChild('content', { static: true })
+  private content: ElementRef;
+  container: HTMLElement;
+
+  // Internal
+  public config: TaskDialogSshConfig = {};
+  public running = false;
+  private subscription: Subscription;
+
+  constructor(
+    private authSessionService: AuthSessionService,
+    @Inject(MAT_DIALOG_DATA) data: TaskDialogSshConfig
+  ) {
+    this.config = data;
+    this.sanitizeConfig();
+    this.socket = io('http://127.0.0.1:1122', { }); // Connect to server Socket.IO    
```

Comment:
How do you plan to make the configuration of the socket configurable so that it can be used by all possible plugins? Since plugins can only make the configuration declaratively, I would be interested in how you want to implement this.

Reply:
physically the address to the server should not change, it should connect via localhost, because of CORS, I would also leave the data for SSH unchanged, only the port could be taken from omv